code/game/objects/items/item_icon_experimental.dm

```diff
@@ -1,9 +1,11 @@
 // This file is an experiment in changing the way item icons are handled.
 // Not really expecting it to work out of the box, so we'll see how it goes
 // with a handful of specific items.
+var/list/if_has_inventory_icon_cache = list()  // for checking if we have special in-inventory HUD state. Cached cause asking icons is expensive
```

Comment:
This would mean that every `/obj/item` initialize a list upon creation, even if never needed

Reply:
This appears to be a global list.

Reply:
it's indoot global